apps/api/src/server/app/inspector/inspector.controller.js

```diff
@@ -59,4 +56,40 @@ const getAppeals = async function(request, response) {
 	return response.send(appealsForResponse);
 };
 
-export { getAppeals };
+const assignAppealsById = async function(userId, appealIds) {
+	const successfullyAssigned = [];
+	const unsuccessfullyAssigned = [];
+	await Promise.all(appealIds.map(async (appealId) => {
+		const appeal = await appealRepository.getById(appealId);
+		if (appeal.userId == undefined && appeal.status == 'available_for_inspector_pickup') {
+			try {
+				const nextState = transitionState({ appealId: appeal.id }, appeal.status, 'PICKUP');
+				await appealRepository.updateById(appeal.id, { status: nextState.value, user: { connect: { id: userId } } });
+				successfullyAssigned.push(appeal.id);
```

Comment:
I think perhaps we should return the assigned/unassigned appeals themselves (using the same appeal interface as the /unassigned endpoint) rather than just the ids . After assigning the appeals, the UI transitions to a screen that displays a table of the successfully assigned appeals, and there'd be no way for the frontend to easily obtain this subset without passing all their data through the express session from the previous screen, which would be a bit ugly.

Reply:
can do!

Reply:
To address in future PR.